src/vega/force_directed_layout.json

```diff
@@ -0,0 +1,141 @@
+{
+    "$schema": "https://vega.github.io/schema/vega/v5.json",
+    "description": "A node-link diagram with force-directed layout, depicting character co-occurrence in the novel Les Misérables.",
```

Comment:
does this remove the need for networkx to run a layout? Is the layout calculated when the @vegaplot is called, or by the plot viewer?

Reply:
So, if force_directed_layout spec gets used, it uses the D3.js library to do the layout (and networkx wouldn't be required for that). It calculates the layout at render-time, so if you're saving to a static file (e.g. png/pdf/etc), it will calculate at save time, but if you save to HTML it will calculate when the html file gets opened in the browser.

That being said, I am not using this spec by default, because networkx still does a much much better job with layouts, especially of large networks than D3. It's included more as an example.